plumbing/api.go

```diff
@@ -291,3 +293,9 @@ func (api *API) DAGGetFileSize(ctx context.Context, c cid.Cid) (uint64, error) {
 func (api *API) DAGCat(ctx context.Context, c cid.Cid) (io.DagReader, error) {
 	return api.dag.Cat(ctx, c)
 }
+
+// DAGImportData adds data from an io stream to the merkledag and returns the
+// Cid of the given data
+func (api *API) DAGImportData(ctx context.Context, data io.Reader) (ipld.Node, error) {
```

Comment:
can you please comment on the side effect of this, which i think that it makes the piece available to fetch from the node via bitswap and keeps a copy in the blockstore though @ZenGround0 knows canonically